### ISSUE TITLE
Badges on player card and profile

### ISSUE THREAD

**Issue by jamesgiu** [jamesgiu/quick-hit#15]
Being a nemesis or victim should be reflected as an icon on someone's player card

Tournament winners should earn badges for their win (maybe the tournament creator can choose an image or have it based on an svg icon, e.g. a golden thumbs up for one tournament)

**Comment by ChrisNeedham24** [comment]
Badge/achievement ideas:

- Clutch Performer: Win 5 games by 2 points or less
- Heart Breaker: Lose 5 games by 2 points or less
- Chicken Dinner: Win a tournament
- Sacked: Beat Frank (my personal favourite)
- On A Roll: Win 3 games in a row
- Unstoppable: Win 5 games in a row
- Ascended: Win 10 games in a row
- Fatality: Win a game without losing a point (to 11 or more)
- Helpless: Lose a game without winning a point (to 11 or more)
- Superior: Reach 1300 ELO

I'll add more if I think of any.

**Comment by jamesgiu** [comment]
LOL at Sacked

**Comment by jamesgiu** [comment]
I'd love to work on this ticket, maybe tonight/tomorrow??

**Comment by jamesgiu** [comment]
Rancher: Have a 5 game winning streak against someone
Boar Farmer: Have a 10 game winning streak against someone
It Ain't Much But it's Honest Work: Have a 15 game winning streak against someone
Heartless: Beat your current Victim 5 times in a row

**Comment by jamesgiu** [comment]
Question: can you get the same achievement twice? E.g. if you get it against 2 different players?

**Comment by jamesgiu** [comment]
With the amount of calculations required, this'll definitely be a backend thing I think (e.g. checking for triggers and recording the achievement in the DB after a game has been entered)

**Comment by ChrisNeedham24** [comment]
> Question: can you get the same achievement twice? E.g. if you get it against 2 different players?

I think it's probably not worth having it more than once, but that's just my opinion. We can gauge opinions tomorrow I suppose.

**Comment by jamesgiu** [comment]
Maybe the achievement text can record the player who was involved in triggering it 

"Boar Farmer

10 win streak against jjg"

**Comment by ChrisNeedham24** [comment]
Just While I'm Deploying: Play 50 games
Raise The Bat: Play 100 games
Need A Charge Code: Play 200 games

**Comment by ChrisNeedham24** [comment]
Cellar Dweller: Drop to 1100 ELO or lower
Drunk On ELO: Gain 100 or more ELO in one happy hour
Mistakes Were Made: Lose 100 or more ELO in one happy hour
Shaking Hands: Win a happy hour game to 11 or more by 2 points
Who Said It Was Happy?: Lose a happy hour game to 11 or more by 2 points
A Little Embarrassing: Lose to a player with an ELO of 200+ less than you
Should Have Put Money On It: Beat a player with an ELO of 200+ more than you

**Comment by jamesgiu** [comment]
These are amazing

**Comment by ChrisNeedham24** [comment]
Open Season: be the victim for 5 other players
Try As You Might: be the nemesis for 5 other players

**Comment by jamesgiu** [comment]
Might work on this tonight

**Comment by jamesgiu** [comment]
First cut is in via https://github.com/jamesgiu/quick-hit/pull/54

Issues:
* match id or data (e.g. scores) isn't stored in the badge so you can't see further match details 
* since it checks your achievements upon each new entered match, if you have a huge backlog of matches from before this update, they'll be attributed to the next match you play, rather than the historical match

**Comment by ChrisNeedham24** [comment]
We may have some issues here, at least a couple of these achievements are being granted without having actually achieved them.

**Comment by jamesgiu** [comment]
Yeah something weird is going on - will have a look tonight if I can (the code where they live is pretty nice to expand on though if you felt like taking a squizz too)

**Comment by ChrisNeedham24** [comment]
Well, the issue for Cellar Dweller is that is it's checking for an ELO of >= 1100, when it should be <= 1100. I believe the issues with the other achievements could be to do with your use of `==` rather than `===` for comparisons.

**Comment by jamesgiu** [comment]
#latenightcommits hahah

**Comment by jamesgiu** [comment]
I'll try and sort those out tonight, also thanks for fucking off those ideas!

**Comment by jamesgiu** [comment]
I have no idea how that last comment had "fucking off" instead of "ticking off" - I had to do like 3 takes of it to believe it. Jesus

**Comment by ChrisNeedham24** [comment]
Imma be honest, I was a little confused too lol

**Comment by jamesgiu** [comment]
Bloody hell - table tennis is serious business

**Comment by jamesgiu** [comment]
Horatio: Play 20 games or more and have a winrate >= 80%

Getting Better (Can't Get No Worse): Play 20 games or more and have a winrate <= 50%

**Comment by ChrisNeedham24** [comment]
Spread the love: beat every other active player
ELO booster: lose to every other active player

**Comment by ChrisNeedham24** [comment]
Was going to take a look at this but decided it was all too hard
